sphinx/search/__init__.py

```diff
@@ -439,15 +439,31 @@ def context_for_searchtool(self) -> Dict[str, Any]:
             js_splitter_code = self.js_splitter_code
 
         return {
-            'search_language_stemming_code': self.lang.js_stemmer_code,
+            'search_language_stemming_code': self.get_js_stemmer_code(),
             'search_language_stop_words': jsdump.dumps(sorted(self.lang.stopwords)),
             'search_scorer_tool': self.js_scorer_code,
             'search_word_splitter_code': js_splitter_code,
         }
 
-    def get_js_stemmer_rawcode(self) -> str:
+    def get_js_stemmer_rawcode(self) -> List[str]:
```

Comment:
We need to keep the compatibility of the interface. How about adding a new method `get_js_stemmer_rawcodes()` (plural form) instead? Or, how about checking the type of `get_js_stemmer_rawcode()` in HTML builder?

Reply:
I did not realize this is actually public API. Now I reverted this method to its previous code, and I am just copying `base-stemmer.js` in HTML builder, if it exists. Is this fine?

Reply:
In the past, we received some claimes for changing non-public APIs. So I'd like to change the interface explicitly as possible.

I think HTML builder should not know about the filename of stemmers. So it's better to change to your first proposal.

* Add a new method `get_js_stemmer_rawcodes()`
* Make an old method `get_js_stemmer_rawcode()` empty (returns None)
* HTML builder calls both old `get_js_stemmer_rawcode()` and new `get_js_stemmer_rawcodes()` to copy stemmer files.

This does not breaks compatibility. And we can deprecate the old one in the next step.